examples/quickstart/jupyter-notebooks/notebooks/03-query/04-UnionOperations.ipynb

```diff
@@ -0,0 +1,644 @@
+   "metadata": {},
+   "source": [
+    "This is what's known as a [top-level](https://druid.apache.org/docs/latest/querying/sql.html#top-level) `UNION` operation: each set of results was gathered individually, one after the other, and the list of results concatenated.\n",
+    "\n",
+    "Notice that these results are not in order by time – even though the individual sets did `ORDER BY` time. Druid has simply concatenated the two result sets together.\n",
+    "\n",
+    "The following cell contains another valid use of `UNION ALL`, where some filtering and `GROUP BY` operations have been added. Notice there is a mix of this being done in the sets themselves and as part of the top level `UNION`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3514ad80-cbe6-4563-8b7b-9d48f8b9a508",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "WITH\n",
+    "set1 AS (\n",
+    "    SELECT \"Reporting_Airline\", \"Distance\"\n",
+    "    FROM \"On_Time_Reporting_Carrier_On_Time_Performance_(1987_present)_2005_11\"\n",
+    "    WHERE DATE_TRUNC('HOUR', __time) = TIMESTAMP '2005-11-01 11:00:00'\n",
+    "  ),\n",
+    "set2 AS (\n",
+    "    SELECT\n",
+    "        \"Reporting_Airline\",\n",
+    "        COUNT(*) AS \"Frights\",\n",
+    "        MAX(Distance) AS \"Lengthiest\",\n",
+    "        MIN(Distance) AS \"Shortest\"\n",
+    "    FROM \"On_Time_Reporting_Carrier_On_Time_Performance_(1987_present)_2005_11\"\n",
+    "    WHERE DATE_TRUNC('HOUR', __time) = TIMESTAMP '2005-11-02 11:00:00'\n",
+    "    AND \"Reporting_Airline\" LIKE 'A%'\n",
+    "    GROUP BY 1\n",
+    "  )\n",
+    "\n",
+    "SELECT \"Reporting_Airline\",\n",
+    "        COUNT(*) AS \"Frights\",\n",
+    "        MAX(Distance) AS \"Lengthiest\",\n",
+    "        MIN(Distance) AS \"Shortest\" from set1\n",
+    "  WHERE \"Reporting_Airline\" LIKE 'AA'\n",
+    "  GROUP BY 1\n",
+    "UNION ALL\n",
+    "SELECT * from set2\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1beb4ad7-93cf-4922-9f57-d7b9739fea4a",
+   "metadata": {},
+   "source": [
+    "Optionally, run the next cell – it shows the precise `EXPLAIN PLAN` for the query. You can see there are two `query` exeuction plans, one for each of our queries, and that Druid's planning process has taken time to optimize how the query above will actually execute."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97934da2-17d1-4c91-8ae3-926cc89185c1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(json.dumps(json.loads(sql_client.explain_sql(sql)['PLAN']), indent=2))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "800add1a-d459-4796-b974-b2f094db417f",
+   "metadata": {},
+   "source": [
+    "The next cell contains two result sets: `set1` provides some statistics for the 1st November, `set2` for the 2nd November.\n",
```

Comment:
Could you label in the results which are from set1 and which are from set2? It would also help to clarify which set this statement applies to, since one set of results has them in the expected order
>Why are the longest and shortest the wrong way around?!
<img width="341" alt="image" src="https://github.com/apache/druid/assets/7747997/71ef5054-0e93-490f-ad88-b3ed3694f716">

Reply:
I've updated the text here so that it better highlights what happens and why – and why the following SQL fixes it.